File: src/main/java/aurocosh/divinefavor/common/custom_data/player/data/presence/energetic/EnergeticPresenceData.java
package aurocosh.divinefavor.common.custom_data.player.data.presence.energetic;

import aurocosh.divinefavor.common.config.common.ConfigPresence;
import aurocosh.divinefavor.common.lib.SimpleCounter;

// The default implementation of the capability. Holds all the logic.
public class EnergeticPresenceData {
    private final SimpleCounter counter;

    public EnergeticPresenceData() {
        counter = new SimpleCounter(ConfigPresence.energeticPresence.timeToRunOnWater);
    }

    public boolean count() {
        return counter.count();
    }

    public void reset() {
        counter.reset();
    }

    public int getCount() {
        return counter.getCount();
    }

    public void setCount(int count) {
        counter.setCount(count);
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/custom_data/player/data/presence/energetic/EnergeticPresenceData.kt
package aurocosh.divinefavor.common.custom_data.player.data.presence.energetic

import aurocosh.divinefavor.common.config.common.ConfigPresence
import aurocosh.divinefavor.common.lib.SimpleCounter

// The default implementation of the capability. Holds all the logic.
class EnergeticPresenceData {
    private val counter: SimpleCounter = SimpleCounter(ConfigPresence.energeticPresence.timeToRunOnWater)

    var count: Int
        get() = counter.count
        set(count) {
            counter.count = count
        }

    fun count(): Boolean {
        return counter.count()
    }

    fun reset() {
        counter.reset()
    }
}
